text/0000-deprecation.md

```diff
@@ -0,0 +1,105 @@
+- Feature Name: Public Stability
+- Start Date: 2015-09-03
+- RFC PR: 
+- Rust Issue: 
+
+# Summary
+
+This RFC proposes to allow library authors to use a `#[deprecated]` attribute,
+with `since="`(version)`"`, `reason="`(free text)`"` and 
+`surrogate="`(text or surrogate declaration)`"` fields. The compiler can then
+warn on deprecated items, while `rustdoc` can document their deprecation
+accordingly. 
+
+# Motivation
+
+Library authors want a way to evolve their APIs; which also involves 
+deprecating items. To do this cleanly, they need to document their intentions 
+and give their users enough time to react.
+
+Currently there is no support from the language for this oft-wanted feature
+(despite a similar feature existing for the sole purpose of evolving the Rust
+standard library). This RFC aims to rectify that, while giving a pleasant
+interface to use while maximizing usefulness of the metadata introduced.
+
+# Detailed design
+
+Public API items (both plain `fn`s, methods, trait- and inherent 
+`impl`ementations as well as `const` definitions, type definitions, struct
+fields and enum variants) can be given a `#[deprecated]` attribute.
+
+This attribute *must* have the `since` field, which contains the exact
+version of the crate that deprecated the item, as defined by Cargo.toml 
+(thus following the semver scheme). It makes no sense to put a version number 
```

Comment:
I think this field should just be opaque text. What benefit do we get from validating semver correctness (or whatever) at the language level? It would force rustc to understand cargo's behaviour and read Cargo.toml's or whatever.